packages/pigeon/test/java_generator_test.dart

```diff
@@ -199,4 +199,34 @@ void main() {
     expect(code, contains('doSomething(Reply<Output>'));
     expect(code, contains('channel.send(null'));
   });
+
+  test('gen list', () {
+    final Root root = Root(apis: <Api>[], classes: <Class>[
+      Class(
+          name: 'Foobar',
+          fields: <Field>[Field(name: 'field1', dataType: 'List')]),
+    ]);
+    final StringBuffer sink = StringBuffer();
+    final JavaOptions javaOptions = JavaOptions();
+    javaOptions.className = 'Messages';
+    generateJava(javaOptions, root, sink);
+    final String code = sink.toString();
+    expect(code, contains('public static class Foobar'));
+    expect(code, contains('private ArrayList field1;'));
```

Comment:
nit: I suggest checking in the output code and then failing the test if there's a diff :)

Reply:
Like golden image tests?  Interesting, i'll keep it in mind.

Reply:
Rather than creating a file infrastructure, just dump the expected output string in text in this file like most of our other tests

Reply:
For the unit tests, the compact and to the point assertions are valuable.  I think I'd prefer to use those as a different kind of test.  Right now I have 3 kinds of tests covering the code:
1) unit tests that use things like regular expressions to test specific settings
2) compilation tests which make sure generated code compiles in different scenarios
3) integration tests which use the generated code

I could see there being something like the compilation tests that checks against predetermined output.  It doesn't seem to fill any gap much other than making sure the output code is pretty which is the lowest priority.